containerfixture/containerfixture.go

```diff
@@ -0,0 +1,133 @@
+package containerfixture
+
+import (
+	"bytes"
+	"fmt"
+	"math/rand"
+	"os/exec"
+	"path/filepath"
+	"runtime"
+	"strings"
+	"testing"
+	"text/template"
+	"time"
+)
+
+const waitTimeoutSeconds = int64(120)
+const containerBuildAttempts = 3
+
+var r = rand.New(rand.NewSource(time.Now().UnixNano()))
+
+type DeploymentInfo struct {
+	Yaml    string
+	PodName string
+	WebPort string
+	ApiPort string
+}
+
+// Generates a random dynamic port number
+func getRandomPort() string {
+	minPort := 49152
+	maxPort := 65535
+	randomPort := fmt.Sprintf("%d", r.Intn(maxPort-minPort)+minPort)
+	return randomPort
+}
+
+// Modify the yaml manifest to allow running it in parallel
+// as many times as needed with dynamic values for its name and ports
+func ApplyDynamicDeploymentTemplate(yaml string) DeploymentInfo {
```

Comment:
if this is a "dynamic deployment" template, what would be a "static" one? also is this a template of a deployment? deployment to where? a production cluster? Lets try to be specific about what we are doing:

```suggestion
func GenerateFixturePodYaml(template string) FixtureInfo {
```

Reply:
fixed